src/controllers/PartialUserFormVerifyController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+
+namespace Firesphere\PartialUserforms\Controllers;
+
+use Firesphere\PartialUserforms\Forms\PasswordForm;
+use Firesphere\PartialUserforms\Models\PartialFormSubmission;
+use Page;
+use PageController;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\UserForms\Model\UserDefinedForm;
+
+/**
+ * Class \Firesphere\PartialUserforms\Controllers\PartialUserFormVerifyController
+ *
+ */
+class PartialUserFormVerifyController extends PageController
+{
+
+    /**
+     * @var array
+     */
+    private static $allowed_actions = [
+        'getForm'
+    ];
+    /**
+     * @var PartialFormSubmission
+     */
+    protected $partialFormSubmission;
+
+    /**
+     * @var PasswordForm
+     */
+    protected $form;
+
+    /**
+     * @return PartialUserFormVerifyController|void
+     * @throws HTTPResponse_Exception
+     */
+    public function init()
+    {
+        parent::init();
+        $session = $this->getRequest()->getSession();
+        // Set the session if the last session has expired
+        if (!$session->get(PartialSubmissionController::SESSION_KEY)) {
+            return $this->httpError(404);
+        }
+
+        /** @var PartialFormSubmission $partial */
+        $partial = PartialFormSubmission::get()->byID($session->get(PartialSubmissionController::SESSION_KEY));
```

Comment:
Since `$session->get(PartialSubmissionController::SESSION_KEY)` is used twice, can we assign it to a variable?

Reply:
👍